### PR TITLE
feat: add fixes for Hapi test suites with non-zero shard or realm values

### PR DESCRIPTION
**Pull request by MiroslavGatsanoga**
**Description**:

Fixed failing Hapi tests for non-zero shard/realm values in the following packages:
- [X] util
  - [X] UtilPrngSuite
- [X] validation
  - [X] StreamValidationTest (tested with SubmitMessageSuite)
  - [X] LogValidationTest (tested with SubmitMessageSuite)
- [X] regression
  - [X] TargetNetworkPrep
  - [X] MixedOpsNodeDeathReconnectTest
  - [X] DabEnabledUpgradeTest
- [X] schedule 
- [X] staking 
- [X] throttling 
  - [X] systemAccountUpdatePrivilegesAsExpected
  - [X] checkBalanceQps
- [X] token 
  - [X] TokenCreateSpecs 
  - [X] TokenManagementSpecs
  - [X] TokenManagementSpecsStateful

**Related issue(s)**:

Fixes https://github.com/hashgraph/hedera-services/issues/17681

**Notes for reviewer**:

 - `utils` package doesn't contain Hapi tests, so nothing to fix there
 - There are some tests failing because they use ContractCreate transactions, those should be fixed as part of work done by HSCS team:
    - AddressAliasIdFuzzing
    - CompletedHollowAccountOperationsFuzzing
    - HollowAccountCompletionFuzzing
    - UmbrellaRedux
    - stakingMetadataUpdateIsRewardOpportunity
    - txOverGasLimitThrottled
    - precompileNftMintsAreLimitedByConsThrottle
    - contractInfoQueriesAsExpected
    - associatedContractsMustHaveAdminKeys
    - dissociateHasExpectedSemanticsForDissociatedContracts
    - expiredAndDeletedTokensStillAppearInContractInfo
    - cannotGiveNftsToDissociatedContractsOrAccounts
    - cannotSendFungibleToDissociatedContractsOrAccounts
    
**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [X] Tested (unit, integration, etc.)

### CODE REVIEW

**Review by petreze — APPROVED** [review]
LGTM

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/blocks/impl/BlockImplUtils.java [hunk 1]
```diff
@@ -144,31 +144,45 @@ public static int stateIdFor(@NonNull final String serviceName, @NonNull final S
...
-                        default -> UNKNOWN_STATE_ID;
-                    };
+                    case "FileService" -> {
+                        if ("FILES".equals(stateKey)) {
+                            yield STATE_ID_FILES.protoOrdinal();
+                        } else if (stateKey.matches(
+                                "UPGRADE_DATA\\[FileID\\[shardNum=\\d, realmNum=\\d, fileNum=150]]")) {
+                            yield STATE_ID_UPGRADE_DATA_150.protoOrdinal();
+                        } else if (stateKey.matches(
+                                "UPGRADE_DATA\\[FileID\\[shardNum=\\d, realmNum=\\d, fileNum=151]]")) {
```
Comment (vtronkov) [inline hedera-node/hedera-app/src/main/java/com/hedera/node/app/blocks/impl/BlockImplUtils.java:154]: **nit**: maybe you can extract the string as a constant:
```
private static final IntFunction<String> UPGRADE_DATA_FILE_FORMAT = n -> String.format(
    "UPGRADE_DATA\\[FileID\\[shardNum=\\d, realmNum=\\d, fileNum=%s]]", n);
```
Then you do `UPGRADE_DATA_FILE_FORMAT.apply()` for all cases.

File: hedera-node/test-clients/src/main/java/com/hedera/services/bdd/junit/hedera/AbstractNode.java [hunk 2]
```diff
@@ -101,11 +101,19 @@ public Path getExternalPath(@NonNull final ExternalPath path) {
             case RECORD_STREAMS_DIR -> workingDir
                     .resolve(DATA_DIR)
                     .resolve(RECORD_STREAMS_DIR)
-                    .resolve("record0.0." + getAccountId().accountNumOrThrow());
+                    .resolve(String.format(
+                            "record%d.%d.%d",
+                            getAccountId().shardNum(),
+                            getAccountId().realmNum(),
+                            getAccountId().accountNumOrThrow()));
```
Comment (vtronkov) [inline hedera-node/test-clients/src/main/java/com/hedera/services/bdd/junit/hedera/AbstractNode.java:108]: just a heads-up: I am fixing the same class in my PR: https://github.com/hiero-ledger/hiero-consensus-node/pull/17860. So we might have a merge conflict.

**Review by vtronkov — APPROVED** [review]
LGTM

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/blocks/impl/BlockImplUtils.java (hunk 1 above)
Comment (MiroslavGatsanoga) [reply hedera-node/hedera-app/src/main/java/com/hedera/node/app/blocks/impl/BlockImplUtils.java]: Updated.

**Review by vtronkov — APPROVED** [review]

**Review by vtronkov — APPROVED** [review]
